// Copyright 2019 The Chromium OS Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";
option optimize_for = LITE_RUNTIME;
package tpm_manager;

// These protobuf messages contain the fields which used to reside in
// |cryptohome| and needed by |tpm_manager| once |tpm_managerd| is launched for
// TPM1.2.
// This is extracted and reduced to minumum based on the need of local data
// migration. The source reference for each message definition is documented
// along with the message declaration. Comments of the fields are preserved and
// copied together.

// From "cryptohome/tpm_status.proto: message TpmStatus"
message LegacyTpmStatus {
  reserved 1, 2;
  optional bytes owner_password = 3;
}

// From "cryptohome/attestation.proto: message Delegation"
message LegacyDelegation {
  // The delegate owner blob.
  optional bytes blob = 1;
  // The authorization secret.
  optional bytes secret = 2;
  // Whether this delegate has permissions to call TPM_ResetLockValue.
  optional bool has_reset_lock_permissions = 3;
  reserved 4;
}

// From "cryptohome/attestation.proto: message AttestationDatabase"
message LegacyAttestationDatabase {
  reserved 1 to 5;
  optional LegacyDelegation delegate = 6;
  reserved 7 to 16;
}
